### PR TITLE
Effects of Feature Scaling on a Machine Learning Model

### LINKED ISSUE #4378: [Machine Learning] Effects of Feature scaling on a machine learning model
### Topic Suggestion
Machine Learning
#### Proposed title of article
Effects of Standard scaling on a machine learning model 

#### Proposed article introduction
Datasets that are used for training models in machine learning have unpredictable values that might end up varying from each other in a broad scale. Nimerical values might have big differences amongst themselves while other non numerical values might not be helpful while working with in their original format ie: booleans and gender.

Standard scaling is introduced inorder to solve this challenge. It standardizes the numbers in a manner that makes it easy to compare the values that are out of each others scope and some of those that are not numerical. In this approach, we will first build a prediction model that is not standardised then build one that is standardised and compare the results to note the advantage of implementing standardisation. 

#### Key takeaways 
1. What is Standard scaling
2. How to implement standard scaling
3. Advantages of standard scaling

#### Article quality
I will be approaching the effect of Standard Scaling in a more practical manner by actually creating two models for comparison using their results and I will also be looking at hyperparameter tuning in addition.

#### References
N/A

### PR DESCRIPTION
**Pull request by Sajeyks**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4378

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing. @Sajeyks
With the recent increase in article volume out team has been working diligently to review every article. Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

**Comment by Kayere** [comment]
Plagiarism check done.

**Comment by Sajeyks** [comment]
Hey @Bayler 👋🏾 . Any updates?

**Comment by hectorkambow** [comment]
@Bayler Can we get a status on this PR 🚀  

Thank you

**Comment by Bayler** [comment]
> Hey @Bayler 👋🏾 . Any updates?

Sorry. I must have missed your comment. Yes @Sajeyks Will be done with the review by end of day tomorrow.

cc: @hectorkambow

**Comment by Bayler** [comment]
@Sajeyks A couple of things before I can continue with the review:
- Your tutorial has no hero image.
- Separate file names using a hyphen rather than underscore. Hyphens are good for SEO. Don't forget to change the file names inside your tutorial too.
- Please provide an external link to the full code. This is important for reproducibility of your results. Preferably Google Colab.
- Avoid using the word 'I' in your tutorial. Use either 'we' or 'you'. Remember, the focus is on the reader.

**Comment by Sajeyks** [comment]
> @Sajeyks A couple of things before I can continue with the review:
> 
> * Your tutorial has no hero image.
> * Separate file names using a hyphen rather than underscore. Hyphens are good for SEO. Don't forget to change the file names inside your tutorial too.
> * Please provide an external link to the full code. This is important for reproducibility of your results. Preferably Google Colab.
> * Avoid using the word 'I' in your tutorial. Use either 'we' or 'you'. Remember, the focus is on the reader.
> * Make sure to add a `Table of contents` section.

Hello @Bayler . I think I've already met your 3rd and 4th suggestions already.

**Comment by Sajeyks** [comment]
I made the requested changes @Bayler. Please have a look. Thank You

File: content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+9.[Conclusion](#conclusion)
+
+###  Prerequisites
+We will need :
+1. Fundamental knowledge in Python programming language.
+2. Fundamental knowledge in Machine Learning.
+3. A python IDE. I recommend Jupyter Notebook/Jupyter Lab/google colab.
+
+We will build models to predict if a patient was diagnosed with either disease `M` or `B` using the given data for training.
+The complete code for this tutorial can also be found [here](https://colab.research.google.com/drive/1v_KegBZvxju3DxGcnsTjQcy_6lPIPG_i?usp=sharing)
```
Comment (lcheop) [inline content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md:40]: Here's the 'I' that I was talking about. 

Also, your external link needs to go right above the conclusion. It is good practice to first describe the code them provide the external link at the bottom

Comment (Bayler) [inline content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md:40]: Here's the 'I' that I was talking about. 

Also, your external link needs to go right above the conclusion. It is good practice to first describe the code them provide the external link at the bottom

File: content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+5.[Using Normalization](#3using-normalization)
+
+6.[Effects of the feature scaling](#effects-of-the-feature-scaling)
+
+7.[Choosing between the two scaling techniques](#choosing-between-the-two-scaling-techniques)
+
+8.[Exceptions](#exceptions)
+
+9.[Conclusion](#conclusion)
```
Comment (Bayler) [inline content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md:47]: Use a dash instead of numbers. In addition, please remove the white space, it is unnecessary.

File: content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+```python
+url = 'https://github.com/Sajeyks/Section-dataset1/blob/main/data.csv?raw=true'
+df = pd.read_csv(url,index_col=0)
+```
+#### (ii) Exploring our dataset
+We will use :
+```python
+df.info()
+```
+to check what our dataset consists of:
```
Comment (Bayler) [inline content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md:80]:
```suggestion
We will use `df.info` to check what our dataset consists of:

```python
df.info()
```
Output:
```

File: content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+ ![df.info()](/engineering-education/effects-of-standard-scaling-on-a-machine-learning-model/df-info.JPG)
+
+ As you can see, our dataset entails 32 columns, where most are floats and one contains objects. The first 31 columns are also in a good state hence no need for replacing or removing some rows.
+ We can also use :
+ ```python 
+ df.head(10)
+ ```
+to have a look at the values in the first ten rows of our dataset:
+
+![fd.head(10)](/engineering-education/effects-of-standard-scaling-on-a-machine-learning-model/df-head.JPG)
```
Comment (Bayler) [inline content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md:91]: Please remove the `.JPG` in caps for all the links inside your article.

File: content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+####  (iii) Preparing the dataset
+We will now select a set of attributes that can affect the dependent variables(diagnosis) in our dataset:
+```python
+my_features = ['radius_mean', 'perimeter_mean', 'area_mean', 'compactness_mean', 'concavity_mean', 'concave points_mean',
+               'perimeter_worst', 'area_worst', 'compactness_worst', 'concavity_worst', 'concave points_worst']
+``` 
+Then we will split the data between `y(dependent)` and `X(indipendent)` variables:
```
Comment (Bayler) [inline content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md:99]: That's the wrong spelling `independent`. Always proofread your work.

File: content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+```python
+df1 = pd.read_csv(url,index_col=0)
+# dataframe summary
+df1.head(10)
+```
+Take note of how the values across the columns are distributed in a wide range. ie some are 122.80 and ,others  0.11840 :
+
+![df.head(10)](/engineering-education/effects-of-standard-scaling-on-a-machine-learning-model/df-head.JPG)
+####  (ii) Implementing standard scaling
+In order to scale properly, we will first collect the names of all the numerical columns in a list then use the list to fit  the standard scaler.
```
Comment (Bayler) [inline content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md:155]:
```suggestion
To scale properly, we will first collect the names of all the numerical columns in a list then use the list to fit  the standard scaler.
```

File: content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+for i in col:
+    sc = StandardScaler().fit(df1[[i]])
+    df1[i] = sc.transform(df1[[i]]) 
+``` 
+Using `standard scaler` from `sklearn`, we were able to rescale our features using the `standardization` technique.
+Now if we check our `df1` again :
+```python
+df1.head(10)
+```
+ you will notice how the range between the values has been reduced:
```
Comment (Bayler) [inline content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md:173]: No need to separate the sentence that way. I've shown you an example above. Fix them where necessary.

**Comment by Bayler** [comment]
@Sajeyks A couple of more revisions before we can move on for final review.

File: content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md (hunk 2 above)
Comment (Sajeyks) [reply content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md:47]: Do you mean the lines between ?

Comment (Bayler) [reply content/articles/effects-of-feature-scaling-on-a-machine-learning-model/index.md]: Yes. Those ones

**Comment by Bayler** [comment]
@Sajeyks This is a very educative tutorial. However, the minor revisions tend to mask the beauty of the content inside. As an aspiring peer reviewer, always pay attention to these little details as that's what makes a good peer reviewer. As a peer reviewer, you should be able to detect these mistakes on the first walk through. Go through all the commits we've made so that you can better your next tutorial. Otherwise, I'm satisfied with the changes.

**Comment by Sajeyks** [comment]
> @Sajeyks This is a very educative tutorial. However, the minor revisions tend to mask the beauty of the content inside. As an aspiring peer reviewer, always pay attention to these little details as that's what makes a good peer reviewer. As a peer reviewer, you should be able to detect these mistakes on the first walk through. Go through all the commits we've made so that you can better your next tutorial. Otherwise, I'm satisfied with the changes.

I will take your advice, Thank you! @Bayler

**Comment by hectorkambow** [comment]
> > @Sajeyks This is a very educative tutorial. However, the minor revisions tend to mask the beauty of the content inside. As an aspiring peer reviewer, always pay attention to these little details as that's what makes a good peer reviewer. As a peer reviewer, you should be able to detect these mistakes on the first walk through. Go through all the commits we've made so that you can better your next tutorial. Otherwise, I'm satisfied with the changes.
> 
> I will take your advice. Thank you! @Bayler

very well said!

**Comment by lalith1403** [comment]
Jumping on this one 🚀

**Review by lalith1403 — APPROVED** [review]
